cmd/sync/runner.go

```diff
@@ -183,41 +191,87 @@ func (r *runner) sync(ctx context.Context, srcRegistry, dstRegistry registry.Int
 			Repo: repo,
 		}
 
-		tagsToSync, err := r.processGetTagsJob(ctx, job)
+		select {
+		case <-ctx.Done():
+			return microerror.Mask(ctx.Err())
+		case err := <-processGetTagsJobErrCh:
+			return microerror.Mask(err)
+		case err := <-processRetagJobsErrCh:
+			return microerror.Mask(err)
+		case getTagsJobCh <- job:
+			// Job added.
+		}
+	}
+
+	// Wait for job processing to finish.
+	{
+		close(getTagsJobCh)
+		err = <-processGetTagsJobErrCh
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		close(retagJobCh)
+		err = <-processRetagJobsErrCh
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	}
 
-		for tagIndex, tag := range tagsToSync {
-			job := retagJob{
-				Src: srcRegistry,
-				Dst: dstRegistry,
+	return nil
+}
 
-				ID:   fmt.Sprintf("Repository [%d/%d] = %#q: Tag [%d/%d] = %#q", repoIndex+1, len(reposToSync), repo, tagIndex+1, len(tagsToSync), tag),
-				Repo: repo,
-				Tag:  tag,
-			}
+func (r *runner) processGetTagsJobs(ctx context.Context, jobCh <-chan getTagsJob, resultCh chan retagJob) error {
```

Comment:
The diff is a bit stupid. `processGetTagsJobs` is added. Look only at the added lines below this comment.